### ISSUE TITLE
UI/UX of notifications break most Linux desktops experience, and many games too, probably affect MacOS

### ISSUE THREAD

**Issue by cm-t** [ValveSoftware/steam-for-linux#9226]
#### Your system information

* Steam client version (build number or date): **Feb 14th, 2023, 00:47:47**
* Distribution : **Ubuntu 22.04 LTS** 64bits _(GnomeShell, default environment for a lot of users)_
* Opted into Steam client beta?: **No**
* Have you checked for system updates?: **Yes**
* Affects: 
    * Any **GnomeShell**/Unity/Lomiri/etc distro, 
    * Probably affect all MacOS user 
    * Most Game with decent HUD _(most in-game informations/actions are not top/right but bottom of the screen, think about almost all RTS, MOBA, racing, plateform, MMORPG, …)_ 

#### Please describe your issue in as much detail as possible:
I have checked for this issue in github, but it is always described as a feature request, I want to make sure this is truly a bug, creating for users.a very nasty user experience with the steam product.

Notifications position are breaking the whole user experience in the system it is used. The feature of notification appears in a area the user click for something else.

Notifications should be able to match the desktop default notification position, or if not, it should be user adjustable.

#### Steps for reproducing this issue:

1. Use any application
2. You receive notifications, they will show in top right area.
3. Many application are designed with this in mind.
4. Now run steam
5. Try to continue to use you daily apps
6. Trying to interact in the bottom border of your screen now becomes tricky
7. Every times a notification from **only** steam will ruins your experience
8. For example you try to send you message in your messenger app (or even within steam friend)
9. You think you're going to click on the send message, but nope, a wild notification appears and give you the focus on the contact of the notification you clicked for example.
10. The only solutions users have now are: 
  * disable all steam notifications
  * use a custom skin to be able to adjust this notification position manually in a config file somewhere, you got lucky if you found a decent howTo. _(but you won't have the default skin anymore, could be a breaking with updates)_
  * Keep it like that, and be mad about steam for more than 10years for just a tiny hole in the UI/UX steam notification position. I mean really mad. Some days i become a true master of the notifications, I see it arriving, and show my mastering of being patient to wait that the notification disappear and during that time I can watch my top/right corner and confirm again and again that there were nothing to block me if it was in top/right corner, and then I become even more mad again. Then it disappear, and finally can to whatever I wanted, on my apps, or even when playing in-game _(thinking about hiding my shopping/buybacks HUD in DotA2, a Valve game for example, while it could just have hidden my ping stats in top/right corner, anyway, i disabled in-game notification now, I don't even know if game publisher can adjust steam's notification for in-game notification…)_
11. Those 3 solutions are NOT acceptable

** **
### Notes

As a daily user of Ubuntu + Steam for more than 10 years, I strongly assure that this notification is really annoying.

It really affect the user experience, of their system, maybe of their game, but it surely affect the feeling of the user about steam, and its publisher, Valve.

I hope you got the bottle in the sea o/ 

#### FAQ

- No it is not a feature request
- No it is not a technical bug  
- Yes it is a UI/UX  bug  
- Yes it is a bug

**Comment by CrumblyLiquid** [comment]
I agree that Steam notifications can be sometimes annoying and do break the desktop experience. However, I don't think that making them _match the desktop default notification position_ is the right way to go here.

I would rather add an option to toggle between **_Steam_ notifications** and **native notification** (most likely implemented via `xdg-desktop-portals` - the same way the new file picker is implemented).
This would allow for Steam to support a **wide range of desktops and notification daemons**.

**Comment by RokeJulianLockhart** [comment]
> add an option to toggle between Steam notifications and native notification (most likely implemented via xdg-desktop-portals - the same way the new file picker is implemented).
This would allow for Steam to support a wide range of desktops and notification daemons.

I agree, @CrumblyLiquid, although, consequently, I don't quite understand what you mean by the undermentioned:

> I don't think that making them match the desktop default notification position is the right way to go here.

**Comment by RokeJulianLockhart** [comment]
@cm-t, I think you should rename the title to "Use native notifications", because this issue was rather difficult to find.

**Comment by vic-bay** [comment]
This is not a bug, rather a feature request indeed, but I still agree to notifications thing. 
Starting from 2024, Steam has dropped windows 7 support. Old windows versions didn't have notification system, that's why Steam made its own. Now Steam supports only windows 10+, macos, and linux. All of these system have their notification implementations, and Steam should use them instead of hacky old notification approach. This approach could fulfil other feature requests, like opt-in disabling achievement notifications (because you could just supress all of them in the system settings), notifications history (system implementation already has its history), and anything else that can be offloaded to the OS.

**Comment by EoceneMiacid** [comment]
Gabe is too busy enjoying his jewel-encrusted yachts to care about a silly thing like user convenience.

**Comment by NextRaven10** [comment]
So what about this issue/Feature request? Any updates?

**Comment by EoceneMiacid** [comment]
Valve doesn't display a whole lot of concern for the 'Linux user experience', they only appear to invest development in piling on more and more crowd-pleasing bells and whistles to their already bloated client.

There are thousands of open issues and very few of them seem to receive any kind of attention.

The client needs a **fundamental** rewrite, so it doesn't rely on 32-bit libraries anymore. Their excuse is that this is a requirement to support older 32-bit games, but this is hogwash.

Another big problem is that the client is completely reliant on XWayland, which was meant to be a stopgap to help ease the transition; X is fully considered to be deprecated now (and hey, good riddance if you ask me).

Again, Valve doesn't care.

All modern operating systems have perfectly capable notification systems, but instead Valve imposes theirs, so that it's nice and consistent for them, I guess.

Better yet, instead of a fundamental rewrite, Valve should provide us with a completely UI-less, dependency-free command line client, containing nothing but the core essentials to allow us to download, update and launch games - and have their (entirely optional) GUI sit on top of that.

This would allow them to close tons of open issues overnight.

The frustrating thing is that they already have such a tool - steamcmd. Unfortunately, it lacks the crucial ability to launch games - by design.

They could literally fix this in a single day. Two, testing included. So, why don't they?

If I buy a car, I expect to be handed over the keys after payment, so that I can operate the vehicle on my own terms.

So why do we settle for accomodating the vendor installing an entire garage in our home as the terms for using our purchased goods?

As you may know, the European Commission recently agreed on the Stop Killing Games act. Good start, but I'd say they need to go further and force the likes of Valve to remove the need to impose their fullblown platforms onto our systems.

**Comment by cm-t** [comment]
Just to keep the topic, I'm wish there is a native notification option AND/OR the choice where the steam-made notification appears, for example top corner/right, for default (desktop) steam mode. Im not talking about big picture/steamdeck.

Whatever the solution implemented, i just do not want notification in the bottom right corner, so for the while i just disabled all desktop notifications from steam.

**Comment by johannesrld** [comment]
> I don't think that making them match the desktop default notification position is the right way to go here.

I think the issue is that there isn't really any standard for "notification position" spec that steam can hook into, for the most part applications send their notifications to org.Freedesktop.Notifications and the compositor displays it however it pleases.

**Comment by effleurager** [comment]
> I think the issue is that there isn't really any standard for "notification position" spec that steam can hook into, for the most part applications send their notifications to org.Freedesktop.Notifications and the compositor displays it however it pleases.

It is there – https://specifications.freedesktop.org/notification/latest-single/#hints, but it is only a hint. DEs are not bound to show the notification as specified, and I think it's better that way in some cases – GNOME doesn't have any tray icons, so placing the notification relative to some position is entirely arbitrary and jarring
